### PR TITLE
Added the possibility to import a metashape ply file

### PR DESCRIPTION
**Pull request by simonbethke**
This should allow people to export the pointcloud along with the camera poses from Metashape and migrate them to the nerfstudio format so its digestable by the nerfstudio training.

@brentyi maybe you can take a look at this :)

### CODE REVIEW

File: nerfstudio/process_data/metashape_utils.py [hunk 1]
```diff
@@ -188,10 +191,21 @@ def metashape_to_json(
 
     data["frames"] = frames
 
+    summary = []
+
+    if isinstance(ply_filename, Path):
```
Comment (jb-ye) [inline nerfstudio/process_data/metashape_utils.py:196]: I believe a proper implementation (similar to what we did for colmap) is to save a "applied_transform" in the data dict (e.g. see [this line](https://github.com/nerfstudio-project/nerfstudio/blob/main/nerfstudio/process_data/colmap_utils.py#L479) in colmap utils), and then apply the applied_transform to point cloud (see [this line](https://github.com/nerfstudio-project/nerfstudio/blob/main/nerfstudio/process_data/colmap_utils.py#L690-L692) in colmap case).

Do you have a sample metashape exported data that I can play? I believe I can help make it in the way I suggested.

Comment (simonbethke) [reply nerfstudio/process_data/metashape_utils.py:196]: Sure! I am not at home at the moment, but I can provide data in a few hours

Comment (simonbethke) [reply nerfstudio/process_data/metashape_utils.py:196]: Here is a very small example: https://drive.google.com/file/d/1GQBYRRQO7opLUqQ2u7z-udcxrfC7zHaj/view?usp=sharing

Comment (simonbethke) [reply nerfstudio/process_data/metashape_utils.py:196]: Damn there is at least one misaligned camera... maybe it still can serve this purpose here. Else I could revisit and fix it. I am not good at using metashape yet. Still in trial phase and thinking if I should buy it or not.

File: nerfstudio/scripts/process_data.py [hunk 2]
```diff
@@ -253,6 +258,12 @@ def main(self) -> None:
         if self.eval_data is not None:
             raise ValueError("Cannot use eval_data since cameras were already aligned with Metashape.")
 
+        if isinstance(self.ply, Path):
```
Comment (brentyi) [inline nerfstudio/scripts/process_data.py:261]: and then `if self.ply is not None`? the current condition looks like it will always be `True`

Comment (simonbethke) [reply nerfstudio/scripts/process_data.py:261]: If just were speaking python ^^
It was supposed to be optional. Sure I simply messed up. I will try fixing that

**Comment by jb-ye** [comment]
@simonbethke Could you give me access to your nerfstudio repo/branch, so I can push changes there?

File: nerfstudio/process_data/metashape_utils.py (hunk 1 above)
Comment (jb-ye) [reply nerfstudio/process_data/metashape_utils.py:196]: Can you try my revision here (I built on top of your changes) and verify it works for you? https://github.com/jb-ye/nerfstudio/tree/metashape_splatfacto

**Comment by jb-ye** [comment]
I somehow messed up this PR, see the new PR here: https://github.com/nerfstudio-project/nerfstudio/pull/3122

**Pull request closed without merging**